challenge.rb

```diff
@@ -1,26 +1,34 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.collect do |word|
+    word.capitalize
+  end
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select do |animal|
+    animal == "dog"
+  end
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject do |animal|
+    animal == "dog"
+  end
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.find_all do |animal|
```

Comment:
cool. you can also use `.grep` which takes a regex pattern. it's super handy.
https://ruby-doc.org/core-2.5.0/Enumerable.html#method-i-grep